crowbar_framework/app/models/service_object.rb

```diff
@@ -1190,6 +1193,15 @@ def apply_role(role, inst, in_queue)
           tmprole = RoleObject.find_role_by_name "#{role_name}_remove"
           unless tmprole.nil?
             elem_remove = tmprole.name
+            # save remove intention in #{@bc_name}-databag
+            databag["deployment"][@bc_name]["elements"]["to_remove"] ||= {}
+
+            old_nodes.each do |old_node|
+              @logger.debug "saving #{elem_remove} intention for #{old_node}"
+              databag["deployment"][@bc_name]["elements"]["to_remove"][old_node] ||= []
+              databag["deployment"][@bc_name]["elements"]["to_remove"][old_node] << elem_remove
```

Comment:
@MaximilianMeister I don't understand - the `databag` variable still refers to the databag object, so why do you say it "gets lost"?